src/generators.mjs

```diff
@@ -51,6 +52,16 @@ const createGenerator = markdownInput => {
    * @param {import('./generators/types.d.ts').GeneratorOptions} options The options for the generator runtime
    */
   const runGenerators = async ({ generators, ...extra }) => {
+    try {
+      if (!statSync(extra.output).isDirectory()) {
+        throw new Error('Output is not a directory');
+      }
+    } catch (err) {
+      if (err.code === 'ENOENT') {
+        mkdirSync(extra.output, { recursive: true });
```

Comment:
we are in a async function so we can use the async mkdir it's should be little bit quicker